src/inputgroup/skin.css

```diff
@@ -86,3 +102,77 @@
     }
   }
 }
+
+.spectrum-Datepicker--range {
+  .spectrum-InputGroup-field {
+    &:focus-ring {
+      box-shadow: none !important;
+    }
+    &[disabled] {
+      ~ hr {
+        color: var(--spectrum-textfield-text-color-disabled);
+      }
+    }
+  }
+
+  /* Focus ring */
+  :focus-ring {
+    ~ .spectrum-Datepicker-focusRing {
+      box-shadow: 0 0 0 1px var(--spectrum-dropdown-border-color-key-focus);
+    }
+    &:invalid,
+    &.is-invalid {
+      ~ .spectrum-FieldButton {
+        box-shadow: 0 0 0 1px var(--spectrum-dropdown-border-color-error);
+      }
+      ~ .spectrum-Datepicker-focusRing {
+        box-shadow: 0 0 0 1px var(--spectrum-dropdown-border-color-error);
+      }
+    }
+  }
+
+  &.is-invalid {
+    .spectrum-InputGroup-field {
+      border-color: var(--spectrum-dropdown-border-color-error) !important;
+    }
+    :focus-ring {
```

Comment:
I think this needs to be `&:focus-ring`?

Reply:
In this case, we're creating a selector for `.spectrum-Datepicker-focusRing` following and adjacent to any keyboard focused sibling. `.spectrum-Datepicker--range :focus-ring ~ .spectrum-Datepicker-focusRing`. The either of the `input` elements or the `button` can have `:focus-ring`.

Reply:
Ah ha! Clever. It would be nice to comment about that so future do-gooders don't undo your goodness.